webview-ui/src/components/ViolationIncidentsList.tsx

```diff
@@ -242,17 +185,22 @@ const ViolationIncidentsList: React.FC<ViolationIncidentsListProps> = ({
           </CardHeader>
           <CardExpandableContent>
             <CardBody>
-              {violation.incidents.map((incident) => (
-                <div key={`${incident.uri}-${incident.lineNumber}`}>
-                  {renderIncident(incident, violation)}
-                </div>
+              {Object.entries(groupIncidentsByMsg(violation.incidents)).map(([message, tuples]) => (
```

Comment:
Would prefer to see the transformation done outside of the render method / JSX.

Reply:
Addressed [in this commit](https://github.com/konveyor/editor-extensions/pull/195/commits/47235052d50a29532e29bad878e7bae43925f5a9).